staging/src/k8s.io/client-go/tools/events/event_recorder.go

```diff
@@ -94,9 +96,18 @@ func (recorder *recorderImpl) makeEvent(refRegarding *v1.ObjectReference, refRel
 	if namespace == "" {
 		namespace = metav1.NamespaceDefault
 	}
+	// The referenced Object name may not be a valid name for Events
+	// and fail the creation because of this, so we replace it by
+	// an alphanumeric string.
+	// Ref: https://issues.k8s.io/127594
+	name := fmt.Sprintf("%s.%x", refRegarding.Name, t.UnixNano())
+	if len(apimachineryvalidation.NameIsDNSSubdomain(name, false)) > 0 {
+		name = fmt.Sprintf("%s.%x", base64.RawURLEncoding.EncodeToString([]byte(refRegarding.Name)), t.UnixNano())
```

Comment:
if the involved object name is long, do we need to possibly trim the length of the encoded segment to fit within the max? something like this (completely untested, might be off by one):

```go
suffix := fmt.Sprintf(".%x", t.UnixNano())
prefix := base64.RawURLEncoding.EncodeToString([]byte(refRegarding.Name))
if len(prefix)+len(suffix) > apimachineryvalidation.DNS1123SubdomainMaxLength {
  prefix = prefix[:apimachineryvalidation.DNS1123SubdomainMaxLength-len(suffix)]
}
name = prefix + suffix
```

Reply:
also, I think it's possible for url-base64-encoding to start or end with `-`, or contain _, both of which I think would fail event name validation?

Reply:
> also, I think it's possible for url-base64-encoding to start or end with `-`, or contain _, both of which I think would fail event name validation?

you. make me think hard, but I think is not possible if it is derived from an alphanumeric character https://www.ascii-code.com/ based on the encoding https://datatracker.ietf.org/doc/html/rfc4648#section-5 , but is simpler to implement a trim those chars than trying to resolve that puzzle 😅

Reply:
see latest commit to not use base64 https://github.com/kubernetes/kubernetes/pull/129790#discussion_r1931002836